x-pack/filebeat/input/http_endpoint/config_test.go

```diff
@@ -0,0 +1,60 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package http_endpoint
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_validateConfig(t *testing.T) {
+	testCases := []struct {
+		name      string // Sub-test name.
+		config    config // Load config parameters.
+		wantError error  // Expected error
+	}{
+		{
+			name: "invalid URL",
+			config: config{
+				URL:          "",
+				ResponseBody: `{"message": "success"}`,
+				Method:       http.MethodPost,
+			},
+			wantError: fmt.Errorf("webhook path URL can not be empty"),
+		},
+		{
+			name: "invalid method",
+			config: config{
+				URL:          "/",
+				ResponseBody: `{"message": "success"}`,
+				Method:       "random",
+			},
+			wantError: fmt.Errorf("method must be POST, PUT or PATCH: random"),
+		},
+		{
+			name: "invalid URL",
+			config: config{
+				URL:          "/",
+				ResponseBody: "",
+				Method:       http.MethodPost,
+			},
+			wantError: fmt.Errorf("response_body must be valid JSON"),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.config.URL = ""
```

Comment:
This won't test the behavior of the `validate` struct tags. That executes when config.Unpack() is called. So if the goal of this test was to unit test your Validate function then its good. 

But if you want test that all of the validation associated with this struct works when Unpack is called, then I would change the test to use `Unpack`.

```diff
diff --git a/x-pack/filebeat/input/http_endpoint/config.go b/x-pack/filebeat/input/http_endpoint/config.go
index 92385a33b1..e073302ef1 100644
--- a/x-pack/filebeat/input/http_endpoint/config.go
+++ b/x-pack/filebeat/input/http_endpoint/config.go
@@ -51,8 +51,6 @@ func defaultConfig() config {
 	return config{
 		Method:        http.MethodPost,
 		BasicAuth:     false,
-		Username:      "",
-		Password:      "",
 		ResponseCode:  200,
 		ResponseBody:  `{"message": "success"}`,
 		ListenAddress: "127.0.0.1",
@@ -60,14 +58,6 @@ func defaultConfig() config {
 		URL:           "/",
 		Prefix:        "json",
 		ContentType:   "application/json",
-		SecretHeader:  "",
-		SecretValue:   "",
-		HMACHeader:    "",
-		HMACKey:       "",
-		HMACType:      "",
-		HMACPrefix:    "",
-		CRCProvider:   "",
-		CRCSecret:     "",
 	}
 }
 
@@ -110,10 +100,6 @@ func (c *config) Validate() error {
 		return errors.New("crc.provider is required when crc.secret is defined")
 	}
 
-	if c.URL == "" {
-		return fmt.Errorf("webhook path URL can not be empty")
-	}
-
 	return nil
 }
 
diff --git a/x-pack/filebeat/input/http_endpoint/config_test.go b/x-pack/filebeat/input/http_endpoint/config_test.go
index f5318e214b..8c285422ec 100644
--- a/x-pack/filebeat/input/http_endpoint/config_test.go
+++ b/x-pack/filebeat/input/http_endpoint/config_test.go
@@ -5,27 +5,28 @@
 package http_endpoint
 
 import (
-	"fmt"
 	"net/http"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+
+	confpkg "github.com/elastic/elastic-agent-libs/config"
 )
 
 func Test_validateConfig(t *testing.T) {
 	testCases := []struct {
 		name      string // Sub-test name.
 		config    config // Load config parameters.
-		wantError error  // Expected error
+		wantError string // Expected error
 	}{
 		{
-			name: "invalid URL",
+			name: "empty URL",
 			config: config{
 				URL:          "",
 				ResponseBody: `{"message": "success"}`,
 				Method:       http.MethodPost,
 			},
-			wantError: fmt.Errorf("webhook path URL can not be empty"),
+			wantError: "string value is not set accessing 'url'",
 		},
 		{
 			name: "invalid method",
@@ -34,27 +35,28 @@ func Test_validateConfig(t *testing.T) {
 				ResponseBody: `{"message": "success"}`,
 				Method:       "random",
 			},
-			wantError: fmt.Errorf("method must be POST, PUT or PATCH: random"),
+			wantError: "method must be POST, PUT or PATCH: random",
 		},
 		{
-			name: "invalid URL",
+			name: "invalid ResponseBody",
 			config: config{
 				URL:          "/",
 				ResponseBody: "",
 				Method:       http.MethodPost,
 			},
-			wantError: fmt.Errorf("response_body must be valid JSON"),
+			wantError: "response_body must be valid JSON",
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			tc.config.URL = ""
-			// Execute config validation
-			err := tc.config.Validate()
+			c := confpkg.MustNewConfigFrom(tc.config)
+
+			config := defaultConfig()
+			err := c.Unpack(&config)
 
 			// Validate responses
-			assert.Equal(t, tc.wantError, err)
+			assert.Contains(t, err.Error(), tc.wantError)
 		})
 	}
 }

```

Reply:
Got it..! Thanks Andrew